app/solutions/routes.py

```diff
@@ -37,13 +41,84 @@ def get_general_solutions():
     they click the general solutions menu button. General solutions are ordered based
     on relevance predicted from users personal values.
     """
+    session_id = request.args.get("session-id")
+    if session_id:
+        user_scores = [np.array(get_scores_vector(session_id))]
+    else:
+        user_scores = None
+
+    if user_scores:
+        user_liberal, user_conservative = predict_radical_political(user_scores)
+    else:
+        user_liberal, user_conservative = None, None
+
     try:
         recommended_general_solutions = (
-            SOLUTION_PROCESSOR.get_user_general_solution_nodes()
+            SOLUTION_PROCESSOR.get_user_general_solution_nodes(
+                user_liberal, user_conservative
+            )
         )
         climate_general_solutions = {"solutions": recommended_general_solutions}
         return jsonify(climate_general_solutions), 200
+
     except:
         raise CustomError(
             message="An error occurred while processing the user's general solution nodes."
         )
+
+
+def predict_radical_political(user_scores):
+    """
```

Comment:
Models have been trained on data that has been augmented in different ways. For example, our current best performing model was trained on Ranked data (i.e., data containing the binary pairwise comparisons between the 10 different attributes). Perhaps we should think about implementing those processing steps here, to be applied to user_scores.
If the data is not correctly process to match the training data of the loaded model, .predict() would throw an "operands could not be broadcast together with shapes {}{}" error.